core/EE_System.core.php

```diff
@@ -927,6 +933,25 @@ private function _maybe_brew_regular()
     }
 
 
+    /**
+     * @since $VID:$
+     * @throws Exception
+     */
+    public function loadRouteMatchSpecifications()
+    {
+        if ($this->request->isAdmin() || $this->request->isFrontend() || $this->request->isIframe()) {
+            try {
+                $this->loader->getShared(
+                    'EventEspresso\core\services\route_match\RouteMatchSpecificationManager'
+                );
```

Comment:
> Should ajax requests be included as well 

hmmm... ya maybe I shouldn't have any restrictions on when the `RouteMatchSpecification`s load since they could conceivably be used for detecting specific AJAX routes as well assuming you know the right request params in advance. 

Will remove all restrictions.